ratpack-bearer-auth/src/main/java/st/ratpack/auth/springsec/SpringSecCheckTokenProvider.java

```diff
@@ -37,17 +40,24 @@ public SpringSecCheckTokenProvider(
 			.build();
 
 
-		return httpClient.post(uri, rs -> rs
-			.body(body -> body.type(APPLICATION_X_WWW_FORM_URLENCODED.toString()).text("token=" + token))
-			.redirects(0)
-			.headers(headers -> {
-				headers.add(
-					HttpHeaderNames.AUTHORIZATION,
-					buildBasicAuthHeader(config.getUser(), config.getPassword())
-				);
-				headers.add(HttpHeaderNames.ACCEPT, APPLICATION_JSON);
-			})
-		);
+		return httpClient
+			.post(uri, rs -> rs
+				.body(body -> body.type(APPLICATION_X_WWW_FORM_URLENCODED.toString()).text("token=" + token))
+				.redirects(0)
+				.headers(headers -> {
+					headers.add(
+						HttpHeaderNames.AUTHORIZATION,
+						buildBasicAuthHeader(config.getUser(), config.getPassword())
+					);
+					headers.add(HttpHeaderNames.ACCEPT, APPLICATION_JSON);
+				}))
+			.retry(2, (attempt, ex) -> {
```

Comment:
Does it make sense to be able to specify the number of retries via configuration?

Reply:
I think since this is an open library we should probably pass it in rather than configure, but yes

Reply:
I'm hesitant to allow services to try too many times as well though

Reply:
We would also need to make a change to the public interface of the "TokenProvider" to make that work.. hmmm

Reply:
Made this configurable through the config.